tests/virt_autotest/login_console_as_kvm_or_xen_role.pm

```diff
@@ -0,0 +1,66 @@
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: For openSUSE virtualization test only. login in console, install kvm/xen patterns if needed.
```

Comment:
This does not make any sense to me. Why we don't reuse what we have?

Reply:
Compare to SLE, openSUSE Tumbleweed installs 'server' role instead of 'kvm' or 'xen' role, so the system_role=server is set in the test suite.  check in
http://10.67.129.51/tests/2501#step/system_role/8
I should have explained it at the beginning. I just added the difference in the PR description.
so we have to have a different login way in openSUSE, to install kvm/xen patterns, and get around the failing libvirtd.service loading due to the incorrect location of iptable binaries(not sure bug or not yet).

Reply:
Yes, better to be added in file as comment.

Reply:
I am still not convinced of this duplicate file. The `login_console.pm` should be universal and used everywhere. Please @okurz can you help me here?

Reply:
> Yes, better to be added in file as comment.
> Besides, if it is difficult for you to tell it is a bug or not, you can consult developers for help. As QA, reporting bug is our most important contribution. Do not forget to confirm it after work around it. BTW, it seems to me more like tiny bugs -- libvirt uses ip related tools in a way that is different from what current tumbleweed provides.

It seems not only libvirtd but other tools try to call binaries in /sbin/ directory. I checked PATH seems to be correct. I have not located the problem. I will follow up this issue when we have a complete test run in O3.

Reply:
> I am still not convinced of this duplicate file. The `login_console.pm` should be universal and used everywhere. Please @okurz can you help me here?

the main function of login_console.pm is to handle pxe menu and grub2 menu after first reboot, then login to ssh console. 

- Regarding to pxe menu, this step has already been done by previous step: reconnect_mgmt_console.pm & first_boot.pm in openSUSE installation.

- about grub2 menu. login_console chooses the kvm or xen entry based on needles to bootup. While in openSUSE TW, there is no kvm or xen role selection during installation. So here just normal/default login without entry selection.

login_console_as_kvm_or_xen_role.pm logins to ssh console, zypper installs kvm/xen patterns, sets serial console, makes necessary workarounds. It is quite different with login_console.

Reply:
> I am still not convinced of this duplicate file. The `login_console.pm` should be universal and used everywhere. Please @okurz can you help me here?

I am not sure I am of much help here with my nitpicking ;) Some comments have been raised already that I think should be fixed first, e.g. https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12912/files#r672776757 stating that in openSUSE distributions we do not have such "roles" hence the name at least is misleading. And I think you should address the test workflows in general how a customer would do: Think about a person that just wants to run KVM based VMs on both SLE as well as openSUSE. They would likely not select completely different ways but try to keep the differences to a minimum.